internal/services/network/virtual_hub_connection_data_source.go

```diff
@@ -0,0 +1,159 @@
+package network
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func dataSourceVirtualHubConnection() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceVirtualHubConnectionRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.VirtualHubConnectionName,
+			},
+
+			"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+
+			"virtual_hub_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.VirtualHubName,
+			},
+
+			"virtual_hub_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"remote_virtual_network_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"internet_security_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"routing": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"associated_route_table_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"propagated_route_table": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"labels": {
+										Type:     pluginsdk.TypeSet,
```

Comment:
(as below) in a data source this'll need to be a TypeList

```suggestion
										Type:     pluginsdk.TypeList,
```